engine/gamesys/src/gamesys/components/comp_sprite.cpp

```diff
@@ -112,78 +112,59 @@ namespace dmGameSystem
         SpriteVertex*                   m_VertexBufferData;
         SpriteVertex*                   m_VertexBufferWritePtr;
         dmGraphics::HIndexBuffer        m_IndexBuffer;
+        uint32_t                        m_VertexCount;
+        uint32_t                        m_IndexCount;
         uint8_t*                        m_IndexBufferData;
         uint8_t*                        m_IndexBufferWritePtr;
         uint8_t                         m_Is16BitIndex : 1;
-        uint8_t                         m_UseGeometries : 1;
+        // uint8_t                         m_UseGeometries : 1;
         uint8_t                         m_ReallocBuffers : 1;
     };
 
     DM_GAMESYS_PROP_VECTOR3(SPRITE_PROP_SCALE, scale, false);
-    DM_GAMESYS_PROP_VECTOR3(SPRITE_PROP_SIZE, size, true);
+    DM_GAMESYS_PROP_VECTOR3(SPRITE_PROP_SIZE, size, false);
 
     static const dmhash_t SPRITE_PROP_CURSOR = dmHashString64("cursor");
     static const dmhash_t SPRITE_PROP_PLAYBACK_RATE = dmHashString64("playback_rate");
     static const dmhash_t SPRITE_PROP_ANIMATION = dmHashString64("animation");
 
+    // The 9 slice function produces 16 vertices (4 rows 4 columns)
+    // and since there's 2 triangles per quad and 9 quads in total,
+    // the amount of indices is 6 per quad and 9 quads = 54 indices in total
+    static const uint8_t DM_VERTEX_COUNT_SLICE9 = 16;
+    static const uint8_t DM_INDEX_COUNT_SLICE9  = 9 * 6;
+    // For the legacy version, we produce a single quad with 4 vertices
+    // and 6 indices, 2 triangles per quad and three points each.
+    static const uint8_t DM_VERTEX_COUNT_LEGACY = 4;
+    static const uint8_t DM_INDEX_COUNT_LEGACY  = 6;
+
     static float GetCursor(SpriteComponent* component);
     static void SetCursor(SpriteComponent* component, float cursor);
     static float GetPlaybackRate(SpriteComponent* component);
     static void SetPlaybackRate(SpriteComponent* component, float playback_rate);
 
-    template<typename T>
-    void fillIndices(T* index, uint32_t indices_count) {
-        for(uint32_t i = 0, v = 0; i < indices_count; i += 6, v += 4)
-        {
-            *index++ = v+0;
-            *index++ = v+1;
-            *index++ = v+2;
-            *index++ = v+2;
-            *index++ = v+3;
-            *index++ = v+0;
-        }
-    }
-
-    static void ReAllocateBuffers(SpriteWorld* sprite_world, dmRender::HRenderContext render_context, uint32_t max_sprite_count, uint32_t num_vertices_per_sprite, uint32_t num_indices_per_sprite) {
+    static void ReAllocateBuffers(SpriteWorld* sprite_world, dmRender::HRenderContext render_context) {
         if (sprite_world->m_VertexBuffer) {
             dmGraphics::DeleteVertexBuffer(sprite_world->m_VertexBuffer);
             sprite_world->m_VertexBuffer = 0;
         }
```

Comment:
Q: Why do we need to remove the old vertex buffer here? (and index buffer later on) we upload them when we have made changes anyway, which wipes the data

Reply:
I think it's just force of habit. I.e. thinking of it like a buffer, and not a handle 🤷